src/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureSVN.java

```diff
@@ -255,18 +273,14 @@ private boolean findSourceCodeSVN(HttpMessage originalMessage) throws Exception
 			//svnsourcefileattackmsg.setRequestHeader(this.getBaseMsg().getRequestHeader());
 			sendAndReceive(svnsourcefileattackmsg, false);  //do not follow redirects
 
-			//if we got a 404 or a redirect specifically, then this is NOT a match
-			//note that since we are simply relying on the file existing or not, we 
-			//will not attempt any fuzzy matching. Old school.
-			//this check is necessary, otherwise a recursive scan on nodes in the url path cause lots of false positives.
-			if (	svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.NOT_FOUND &&  //404
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.MOVED_PERMANENTLY && //301
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.FOUND && //302
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.SEE_OTHER	&& //303
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.NOT_MODIFIED && //304
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.USE_PROXY && //305 (306 is currently unused)
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.TEMPORARY_REDIRECT //307
-					) {
+			int attackmsgResponseStatusCode = svnsourcefileattackmsg.getResponseHeader().getStatusCode();
+			//At LOW and MEDIUM ignore all client and server error responses
+			if ((attackStrength==AttackStrength.LOW || attackStrength==AttackStrength.MEDIUM) && 
```

Comment:
Since the logic is the same for all the cases it's better to extract a method (it will be easier to maintain).